processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -760,6 +872,199 @@ public void close() throws IOException
     }
   }
 
+  public static class CursorOffsetHolder
+  {
+    Offset currOffset = null;
```

Comment:
I kept this public since I use the same class in another PR i have open, the dimension handling interface one:

https://github.com/jon-wei/druid/blob/171c870630f07c363240139facc51a6ed24e2f50/processing/src/main/java/io/druid/segment/DimensionColumnReader.java#L98

where the CursorOffsetHolder is provided to an external class for making a DimensionSelector